pandas/core/indexes/multi.py

```diff
@@ -562,6 +562,20 @@ def from_tuples(
             arrs = zip(*tuples)
             arrays = cast(List[Sequence[Hashable]], arrs)
 
+        if len(tuples) != 0:
```

Comment:
this is going to be a considerable perf hit.

can you just look at the arrays?

Reply:
Hi I not able to use arrays. If `tuples =[ ( ), ( ) ]`. 
```
elif isinstance(tuples, list):
     arrays = list(lib.to_object_array_tuples(tuples).T)
```
The arrays will equal to `[ ]` after this section